packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3832,6 +3869,11 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
     for (final _RouteEntry entry in _history) {
       entry.route.changedExternalState();
     }
+
+    // If this is a nested Navigator, handle system backs so that the root
+    // Navigator doesn't get all of them.
+    _handlesBackGestures = widget.handlesBacksWhenNested
+        && Navigator.maybeOf(context, rootNavigator: true) != this;
```

Comment:
This is unfortunate that rootNavigator needs to be set to true...

Reply:
can we do this in activate? `https://api.flutter.dev/flutter/widgets/State/activate.html`
and maybe do a assert make sure this won't change in didChangeDepedencies
I think this is the only possible place this can change, i.e. moving navigator around using globalkey.

Reply:
we also need to update _handlesBackGestures in didUpdateWidget in case the widget.handlesBacksWhenNested changes

Reply:
> This is unfortunate that rootNavigator needs to be set to true...

For performance reasons you mean? The reason I do this is because I need to know if this Navigator is the root Navigator or not. I could do something like this but is it actually any better?

```dart
Element _getParentContext() {
  late final Element parentElement;
  context.visitAncestorElements((Element element) {
    parentElement = element;
    return false;
  });
  return parentElement;
}

bool isRootNavigator() {
  return Navigator.maybeOf(_getParentContext()) == null;
}
```

Reply:
> can we do this in activate? `https://api.flutter.dev/flutter/widgets/State/activate.html` and maybe do a assert make sure this won't change in didChangeDepedencies I think this is the only possible place this can change, i.e. moving navigator around using globalkey.

Done in https://github.com/flutter/flutter/pull/152330/commits/5cb9fc457c713d6662d78f33c6ee3d4694e29465, thanks.